scripts/core/editor3/components/MultipleHighlights.jsx

```diff
@@ -119,70 +119,132 @@ export class MultipleHighlights extends React.Component {
         return state;
     }
 
-    addHighlight(highlightType, highlightData) {
-        // NICE-TO-HAVE: remove associated data when highlights are removed by deleting characters in the editor
+    setHighlightsState(editorState, highlightsState) {
+        // TODO: save to first block
+        this.setState(highlightsState);
+    }
+
+    getHighlightsStates(editorState) {
+        // TODO: get from first block
+        return this.state;
+    }
+
+    saveHighlights({newEditorState, newHighlightsState}, applyChanges) {
+        this.setHighlightsState(newEditorState, newHighlightsState);
+
+        if (applyChanges) {
+            this.props.onHighlightChange(newEditorState, this.exportState());
```

Comment:
We don't need `exportState` anymore since we are storing it all in the first block of the `editorState`